deploy/teardown_resources.sh

```diff
@@ -7,31 +7,60 @@ set -euxo pipefail
 set_namespace $CONJUR_NAMESPACE_NAME
 
 configure_cli_pod
+
+# Helm Chart clean-up
+rm -f conjur.pem
+
 if [ "${DEV}" = "false" ]; then
-  $cli_with_timeout "exec $(get_conjur_cli_pod_name) -- conjur variable values add secrets/test_secret \"supersecret\""
+  pushd ../../../secrets-provider/ci
+else
+  pushd ../secrets-provider/ci
+fi
+
+  find . -type f ! -name '*template.yaml' -delete
+popd
+
+# Delete Helm Chart if already exists
+set_namespace $APP_NAMESPACE_NAME
+if [ "$(helm ls -aq | wc -l | tr -d ' ')" != 0 ]; then
+  helm delete $(helm ls -aq)
 fi
 
+set_namespace $CONJUR_NAMESPACE_NAME
+
+$cli_with_timeout "exec $(get_conjur_cli_pod_name) -- conjur variable values add secrets/test_secret \"supersecret\""
+
 set_namespace $APP_NAMESPACE_NAME
 
 $cli_with_timeout "delete secret dockerpullsecret --ignore-not-found=true"
 
 $cli_with_timeout "delete clusterrole secrets-access-${UNIQUE_TEST_ID} --ignore-not-found=true"
 
+$cli_with_timeout "delete role secrets-provider-role2 --ignore-not-found=true"
```

Comment:
i am not following. what is the `2` here? why do we need 2 roles? why do we need 2 of everything?

Reply:
If you look at test `TEST_ID_18_helm_multiple_provider_multiple_secrets.sh` and `TEST_ID_19_helm_multiple_provider_same_secret.sh` we see that we checking when there are 2 Secrets Providers in the namespace (#18) and when there are 2 Secrets Providers **and** 2 K8s secrets in the namespace. We cannot use the same resource names for both which is why there is two of everything

Reply:
yes, this explains why we need 2 secrets. but do we need 2 roles? 2 rolebindings?